homeassistant/components/transmission/__init__.py

```diff
@@ -156,18 +217,42 @@ async def add_torrent(service: ServiceCall) -> None:
 
     async def start_torrent(service: ServiceCall) -> None:
         """Start torrent."""
+        entry_id: str = service.data[CONF_ENTRY_ID]
+        if (entry := hass.config_entries.async_get_entry(entry_id)) is None or (
+            entry.state is ConfigEntryState.LOADED
+        ):
+            raise HomeAssistantError(
+                f"Config entry {entry_id} is not found or not loaded"
+            )
+        coordinator: TransmissionDataUpdateCoordinator = entry.runtime_data
         torrent_id = service.data[CONF_ID]
         await hass.async_add_executor_job(coordinator.api.start_torrent, torrent_id)
         await coordinator.async_request_refresh()
 
     async def stop_torrent(service: ServiceCall) -> None:
         """Stop torrent."""
+        entry_id: str = service.data[CONF_ENTRY_ID]
+        if (entry := hass.config_entries.async_get_entry(entry_id)) is None or (
+            entry.state is ConfigEntryState.LOADED
+        ):
+            raise HomeAssistantError(
+                f"Config entry {entry_id} is not found or not loaded"
+            )
+        coordinator: TransmissionDataUpdateCoordinator = entry.runtime_data
         torrent_id = service.data[CONF_ID]
         await hass.async_add_executor_job(coordinator.api.stop_torrent, torrent_id)
         await coordinator.async_request_refresh()
 
     async def remove_torrent(service: ServiceCall) -> None:
         """Remove torrent."""
+        entry_id: str = service.data[CONF_ENTRY_ID]
+        if (entry := hass.config_entries.async_get_entry(entry_id)) is None or (
+            entry.state is not ConfigEntryState.LOADED
+        ):
+            raise HomeAssistantError(
+                f"Config entry {entry_id} is not found or not loaded"
+            )
+        coordinator: TransmissionDataUpdateCoordinator = entry.runtime_data
```

Comment:
That's a lot of duplicate code.
You could maybe create a local `_get_coordinator_from_service_data` function instead.